lib/features/assessment/cubit/assessment_cubit.dart

```diff
@@ -0,0 +1,172 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:school_erp/enums/action_type.dart';
+import 'package:school_erp/enums/assessment_type.dart';
+import 'package:school_erp/features/assessment/assessment_service.dart';
+import 'package:school_erp/features/assessment/cubit/assessment_state.dart';
+import 'package:school_erp/features/auth/auth.dart';
+import 'package:school_erp/models/assessment.dart';
+import 'package:school_erp/models/assessment_taker.dart';
+
+class AssessmentCubit extends Cubit<AssessmentState> {
+  final AssessmentService _assessmentService;
+
+  AssessmentCubit({
+    required AssessmentService assessmentService,
+    required AssessmentType assessmentTypeOnCreate,
+    required AuthenticatedUser authUser,
+    Assessment? assessment,
+  })  : _assessmentService = assessmentService,
+        super(AssessmentState.initial(
+          authUser: authUser,
+          assessmentTypeOnCreate: assessmentTypeOnCreate,
+          existingAssessment: assessment,
+        )) {
+    _loadAssessmentTakers();
+  }
+
+  void updateAssessment(Assessment assessment) {
+    emit(
+      state.copyWith(
+          assessment: assessment, status: AssessmentStateStatus.staging),
+    );
+  }
+
+  void addAssessmentTaker() {
+    var assessmentTaker = AssessmentTaker(
+      assessmentId: state.assessment.id ?? '',
+      sectionId: '',
```

Comment:
handle null requirements in object class remove required fields if not actually required. Unsutainable if we duplicate or replicate to another feature